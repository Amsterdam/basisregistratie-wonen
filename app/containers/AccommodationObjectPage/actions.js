import {
  LOAD_NUMMERAANDUIDING_DATA,
  LOAD_NUMMERAANDUIDING_DATA_SUCCESS,
  LOAD_NUMMERAANDUIDING_DATA_FAILED,
  LOAD_PAND_DATA,
  LOAD_PAND_DATA_FAILED,
  LOAD_PAND_DATA_SUCCESS,
  LOAD_OPENBARE_RUIMTE_DATA_SUCCESS,
  LOAD_OPENBARE_RUIMTE_DATA,
  LOAD_OPENBARE_RUIMTE_DATA_FAILED,
  LOAD_KADASTER_OBJECT_DATA_FAILED,
  LOAD_KADASTER_OBJECT_DATA_SUCCESS,
  LOAD_KADASTER_OBJECT_DATA,
  LOAD_PANDLIST_DATA,
  LOAD_PANDLIST_DATA_SUCCESS,
  LOAD_PANDLIST_DATA_FAILED,
  LOAD_LIGPLAATS_DATA,
  LOAD_LIGPLAATS_DATA_SUCCESS,
  LOAD_LIGPLAATS_DATA_FAILED,
  LOAD_VERBLIJFSOBJECT_DATA,
  LOAD_VERBLIJFSOBJECT_DATA_SUCCESS,
  LOAD_VERBLIJFSOBJECT_DATA_FAILED,
  LOAD_BAG_DATA,
  LOAD_BAG_DATA_FAILED,
  LOAD_BAG_DATA_SUCCESS,
  LOAD_DATA_SUCCESS,
  LOAD_DATA_FAILED,
} from './constants';

export const loadDataSuccess = payload => ({
  type: LOAD_DATA_SUCCESS,
  payload,
});

export const loadDataFailed = payload => ({
  type: LOAD_DATA_FAILED,
  payload,
});

export const loadBAGData = payload => ({
  type: LOAD_BAG_DATA,
  payload,
});

export const loadBAGDataFailed = payload => ({
  type: LOAD_BAG_DATA_FAILED,
  payload,
});

export const loadBAGDataSuccess = payload => ({
  type: LOAD_BAG_DATA_SUCCESS,
  payload,
});

export const loadBRKData = id => ({
  type: LOAD_KADASTER_OBJECT_DATA,
  payload: { id },
});

export const loadBRKDataSuccess = payload => ({
  type: LOAD_KADASTER_OBJECT_DATA_SUCCESS,
  payload,
});

export const loadBRKDataFailed = payload => ({
  type: LOAD_KADASTER_OBJECT_DATA_FAILED,
  payload,
});

export const loadNummeraanduidingData = id => ({
  type: LOAD_NUMMERAANDUIDING_DATA,
  payload: { id },
});

export const loadNummeraanduidingSuccess = payload => ({
  type: LOAD_NUMMERAANDUIDING_DATA_SUCCESS,
  payload,
});

export const loadNummeraanduidingFailed = payload => ({
  type: LOAD_NUMMERAANDUIDING_DATA_FAILED,
  payload,
});

export const loadPandData = id => ({
  type: LOAD_PAND_DATA,
  payload: { id },
});

export const loadPandDataSuccess = payload => ({
  type: LOAD_PAND_DATA_SUCCESS,
  payload,
});

export const loadPandDataFailed = payload => ({
  type: LOAD_PAND_DATA_FAILED,
  payload,
});

export const loadPandlistData = id => ({
  type: LOAD_PANDLIST_DATA,
  payload: { id },
});

export const loadPandlistDataSuccess = payload => ({
  type: LOAD_PANDLIST_DATA_SUCCESS,
  payload,
});

export const loadPandlistDataFailed = payload => ({
  type: LOAD_PANDLIST_DATA_FAILED,
  payload,
});

export const loadOpenbareRuimteData = id => ({
  type: LOAD_OPENBARE_RUIMTE_DATA,
  payload: { id },
});

export const loadOpenbareRuimteDataSuccess = payload => ({
  type: LOAD_OPENBARE_RUIMTE_DATA_SUCCESS,
  payload,
});

export const loadOpenbareRuimteDataFailed = payload => ({
  type: LOAD_OPENBARE_RUIMTE_DATA_FAILED,
  payload,
});

export const loadLigplaatsData = id => ({
  type: LOAD_LIGPLAATS_DATA,
  payload: { id },
});

export const loadLigplaatsDataSuccess = payload => ({
  type: LOAD_LIGPLAATS_DATA_SUCCESS,
  payload,
});

export const loadLigplaatsDataFailed = payload => ({
  type: LOAD_LIGPLAATS_DATA_FAILED,
  payload,
});

export const loadVerblijfsobjectData = id => ({
  type: LOAD_VERBLIJFSOBJECT_DATA,
  payload: { id },
});

export const loadVerblijfsobjectDataSuccess = payload => ({
  type: LOAD_VERBLIJFSOBJECT_DATA_SUCCESS,
  payload,
});

export const loadVerblijfsobjectDataFailed = payload => ({
  type: LOAD_VERBLIJFSOBJECT_DATA_FAILED,
  payload,
});
