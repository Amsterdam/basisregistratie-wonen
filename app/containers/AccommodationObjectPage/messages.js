/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  does_not_apply: {
    id: 'boilerplate.containers.AccommodationObjectPage.does_not_apply',
    defaultMessage: 'n.v.t.',
  },
});
