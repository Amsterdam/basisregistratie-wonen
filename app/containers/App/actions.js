import {
  AUTHENTICATE_USER,
  AUTHORIZE_USER,
  SHOW_GLOBAL_ERROR,
  RESET_GLOBAL_ERROR,
  LOGIN,
  LOGOUT,
  LOAD_DATA_SUCCESS,
  LOAD_DATA_FAILED,
  LOAD_DATA_PENDING,
  UNAUTHORIZED,
  UNABLE_TO_FETCH,
} from './constants';

export function authenticateUser(credentials) {
  return {
    type: AUTHENTICATE_USER,
    payload: credentials,
  };
}

export function authorizeUser(credentials) {
  return {
    type: AUTHORIZE_USER,
    payload: credentials,
  };
}

export function showGlobalError(message) {
  return {
    type: SHOW_GLOBAL_ERROR,
    payload: message,
  };
}

export function resetGlobalError() {
  return {
    type: RESET_GLOBAL_ERROR,
  };
}

export function doLogin(domain) {
  return {
    type: LOGIN,
    payload: domain,
  };
}

export function doLogout() {
  return {
    type: LOGOUT,
    payload: null,
  };
}

export const statusPending = payload => ({
  type: LOAD_DATA_PENDING,
  payload,
});

export const statusSuccess = payload => ({
  type: LOAD_DATA_SUCCESS,
  payload,
});

export const statusFailed = payload => ({
  type: LOAD_DATA_FAILED,
  payload,
});

export const statusUnableToFetch = payload => ({
  type: UNABLE_TO_FETCH,
  payload,
});

export const statusUnauthorized = payload => ({
  type: UNAUTHORIZED,
  payload,
});
