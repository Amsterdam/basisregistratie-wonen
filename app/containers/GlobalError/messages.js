/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
});
