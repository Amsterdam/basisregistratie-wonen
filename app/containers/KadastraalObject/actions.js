import { LOAD_DATA, LOAD_DATA_FAILED, LOAD_DATA_NO_RESULTS, LOAD_DATA_SUCCESS } from './constants';

export const loadData = adresseerbaarObjectId => ({
  type: LOAD_DATA,
  payload: { adresseerbaarObjectId },
});

export const loadDataSuccess = payload => ({
  type: LOAD_DATA_SUCCESS,
  payload,
});

export const loadDataFailed = payload => ({
  type: LOAD_DATA_FAILED,
  payload,
});

export const loadDataNoResults = payload => ({
  type: LOAD_DATA_NO_RESULTS,
  payload,
});
