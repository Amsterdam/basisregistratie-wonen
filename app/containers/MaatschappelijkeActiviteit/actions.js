import { LOAD_DATA_FAILED, LOAD_DATA_NO_RESULTS, LOAD_DATA_SUCCESS, LOAD_DATA } from './constants';

export const loadData = maatschappelijkeActiviteitId => ({
  type: LOAD_DATA,
  payload: { maatschappelijkeActiviteitId },
});

export const loadDataSuccess = payload => ({
  type: LOAD_DATA_SUCCESS,
  payload,
});

export const loadDataFailed = payload => ({
  type: LOAD_DATA_FAILED,
  payload,
});

export const loadDataNoResults = payload => ({
  type: LOAD_DATA_NO_RESULTS,
  payload,
});
