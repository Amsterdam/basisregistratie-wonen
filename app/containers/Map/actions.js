import { SEARCH_SELECT } from './constants';

export const searchSelect = payload => ({
  type: SEARCH_SELECT,
  payload,
});
