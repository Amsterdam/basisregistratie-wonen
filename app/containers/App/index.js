import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import AccommodationObjectPage from 'containers/AccommodationObjectPage/Loadable';
import Map from 'containers/MapContainer/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import Footer from 'components/Footer';
import HeaderContainer from 'containers/HeaderContainer';
import GlobalError from 'containers/GlobalError';
import { isAuthenticated } from 'shared/services/auth/auth';

import reducer from './reducer';
import saga from './saga';

import GlobalStyles from '../../global-styles';

export const App = () => (
  <div className="container app-container">
    <GlobalError />
    <div className="container">
      <HeaderContainer />
    </div>
    <div className="content container">
      <Switch>
        <Route exact path="/" component={Map} />
        {!isAuthenticated() && <Redirect to="/" />}
        <Route
          exact
          path="/:natRegId,:adresseerbaarObjectId,:nummeraanduidingId,:openbareRuimteId,:latitude,:longitude/"
          component={AccommodationObjectPage}
        />
        <Route path="" component={NotFoundPage} />
      </Switch>
    </div>
    <div className="container-fluid">
      <Footer />
    </div>
    <GlobalStyles />
  </div>
);

const withReducer = injectReducer({ key: 'global', reducer });
const withSaga = injectSaga({ key: 'global', saga });

export default compose(
  withReducer,
  withSaga,
)(App);
