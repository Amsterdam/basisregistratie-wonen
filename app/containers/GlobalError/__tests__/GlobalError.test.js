import React from 'react';
import { render } from 'react-testing-library';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';
import history from 'utils/history';
import { Provider } from 'react-redux';

import * as appActions from 'containers/App/actions';
import GlobalErrorContainer, { GlobalErrorComponent as GlobalError } from '..';
import messages from '../../../translations/nl.json';
import configureStore from '../../../configureStore';

const store = configureStore({}, history);
const intl = {
  formatDate: ({ id }) => id,
  formatMessage: obj => (obj ? obj.id : ''),
  formatNumber: ({ id }) => id,
  formatPlural: ({ id }) => id,
  formatRelative: ({ id }) => id,
  formatTime: ({ id }) => id,
  formatHTMLMessage: ({ id }) => id,
  now: () => Date.now,
};

const renderGlobalError = (props = {}) => (
  <Provider store={store}>
    <IntlProvider locale="nl" messages={messages}>
      <GlobalErrorContainer intl={intl} {...props} />
    </IntlProvider>
  </Provider>
);

describe('<GlobalError />', () => {
  const { container, rerender } = render(renderGlobalError());

  it('should render showing no error by default', () => {
    expect(container.firstChild).toBeNull();
  });

  it('should render showing an error when defined', () => {
    const props = {
      error: true,
      errorMessage: 'MOCK_ERROR',
    };

    rerender(renderGlobalError(props));

    expect(container.firstChild).toMatchSnapshot();
    expect(document.getElementsByTagName('button')).toHaveLength(1);
  });

  it('should close the error when the button is clicked', () => {
    const resetGlobalErrorSpy = jest.spyOn(appActions, 'resetGlobalError');
    const tree = mount(
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages}>
          <GlobalError />
        </IntlProvider>
      </Provider>,
    );

    const closeButton = tree.find('button').first();
    closeButton.simulate('click');

    expect(resetGlobalErrorSpy).toHaveBeenCalled();
  });

  it('should render a feedback button', () => {
    const props = {
      error: true,
      errorMessage: 'MOCK_ERROR',
      onClose: jest.fn(),
    };
    const renderedComponent = mount(renderGlobalError(props));
    expect(renderedComponent).toMatchSnapshot();

    renderedComponent.find('button').simulate('click');
    expect(props.onClose).toHaveBeenCalled();
    expect(renderedComponent).toMatchSnapshot();
  });
});
