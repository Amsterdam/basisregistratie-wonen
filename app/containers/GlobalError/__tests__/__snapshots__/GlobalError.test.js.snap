// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/GlobalError should render showing an error when defined 1`] = `
<div
  class="GlobalError__ErrorWrapper-igrx8k-0 kuoiLJ no-print cf"
>
  <div
    class="GlobalError__ErrorContainer-igrx8k-1 dDcAQy"
  >
    <p
      class="GlobalError__P-igrx8k-2 fGvvtl"
    >
      registraties.unauthorized
    </p>
    <button
      class="GlobalError__Button-igrx8k-3 dCWsfn"
      type="button"
    >
      <svg
        fill="#fff"
        focusable="false"
        viewBox="0 0 32 32"
        width="20"
      >
        <path
          d="M29.333 5.47L26.53 2.668 16 13.187 5.47 2.666 2.668 5.47 13.187 16 2.666 26.53l2.804 2.803L16 18.813l10.53 10.52 2.803-2.804L18.813 16z"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="GlobalError__Label-igrx8k-4 fSyAoo"
      >
        registraties.close
      </span>
    </button>
  </div>
</div>
`;

exports[`containers/GlobalError should render showing no error by default 1`] = `<div />`;
